src/main.rs

```diff
@@ -1,20 +1,23 @@
 use lalrpop_util::lalrpop_mod;
 
 lalrpop_mod!(sysy);
+
 pub mod ast;
 pub mod error;
-pub mod ir_gen;
+pub mod gen;
+
+pub use gen::asm as asm_gen;
+pub use gen::ir as ir_gen;
 
```

Comment:
```suggestion
use gen::asm as asm_gen;
use gen::ir as ir_gen;
```

Remove `pub` as no need here.